### DISCUSSION TITLE
CoreDNS batching

### DISCUSSION THREAD

**Discussion by jaswanthikolla** [coredns/coredns#6033] [General]
When the a certain DNS entry is evicted, A request queries for the same DNS entry at T1 and response is inserted into cache at T2. What happens to all the  concurrent DNS queries requesting the same domain from the time T1 to T2? Are they batched or duplicate DNS queries to the upstream?

**Comment by chrisohaver** [comment]
Duplicate DNS queries are sent to the upstream.

**Comment by mattjamesaus** [comment]
Is this still the case? We've noticed some issues with Route53 where i *think* we're getting a cache expiry and then under high load (10s of thousands of reqs per min) getting rate limited. Possibly because of the shear volume of requests that make it through during that window.

Is there any appetite to make de-duplication of in-flight requests when using cache plugin a thing?
